linux/LinuxProcessList.c

```diff
@@ -1609,7 +1609,7 @@ static inline void LinuxProcessList_scanMemoryInfo(ProcessList* this) {
                (variable) = parsed_;                                         \
             }                                                                \
             break;                                                           \
-         }
+         } else (void) 0
```

Comment:
What's wrong with this? You can't have an else statement after a semicolon.

Reply:
But one can forget the semicolon:
```c
if (condition)
    tryRead("MemTotal:", totalMem)
else
    printf("Condition not met\n");
```
here the `else` branch will not be associated to `if (condition)` but to the `if` condition inside the macro.

Reply:
Previously you could write:

```c
tryRead("foo", &foo)
else 
tryRead("bar", &bar)
else
tryRead("baz", &baz)
else {
   // Really not intended …
}
```

This change ensures you have to follow up with a `;` as if this was a normal "function call" …

Reply:
@cgzones The `if` already breaks when the condition is true. So concatenating an `else` after this macro won't make sense. And I don't think anyone will use a macro with a dangling else like in your example (it's a control affecting macro, and according to the pattern it's only intended to be used inside a `case` statement).

The only compelling reason to me is to force a semicolon after the macro.
I would add a comment explaining the purpose of the null statement:
```suggestion
         } else (void) 0 /* Require a ";" after the macro use. */
```